### ISSUE TITLE
Some instructions in README seem unnecessary

### ISSUE THREAD

**Issue by houmie** [SasukeFreestyle/XTLS-Iran-Reality#13]
Hello,

Thanks for the amazing tutorial.  There is one issue that I found with your instructions.

In this section below you say to create a folder called xray, and download the iran.dat file.
But remember you are doing this on the server, not client.

What is the purpose behind this? Because the server isn't using the iran.dat file.

```
mkdir xray

Change directory to the newly created xray folder.

cd xray/

Download the latest geoasset file for blocking Iranian websites.

wget https://github.com/bootmortis/iran-hosted-domains/releases/latest/download/iran.dat
```

Thanks

**Comment by SasukeFreestyle** [comment]
While the purpose of iran.dat on clients is to use Iranian IP for Iranian websites the purpose of the server is to block all Iranian websites listed in iran.dat, so the server does not communicate with those sites.


The server uses iran.dat file here in the config, as a block rule.
https://github.com/SasukeFreestyle/XTLS-Iran-Reality/blob/main/config.json#L1980

**Comment by houmie** [comment]
Ah I think I understand. Interesting.  Xray expects the iran.dat file to be in the same folder as xray binary defined in Service:

```
ExecStart=/home/USERNAME/xray/xray run -config /home/USERNAME/xray/configs.json
```

So in this case iran.dat file will be expected under `/home/USERNAME/xray/iran.dat` so that the block rule `"ext:iran.dat:ir",` can see the file and read from it. Am I correct?

**Comment by SasukeFreestyle** [comment]
Yes. Exactly.

Xray will look for .dat files in the folder it runs in and
```
/usr/local/share/xray
/usr/share/xray
```
More info here https://xtls.github.io/en/config/features/env.html
And
https://xtls.github.io/en/config/routing.html#ruleobject
>Load domains from a file: Formatted as "ext:file:tag", where the file is stored in the [resource directory](https://xtls.github.io>/en/config/features/env.html#resource-file-path) and has the same format as geosite.dat. The tag must exist in the file.

Off-topic:
I just realized after many months I had a typo in the service file.
`ExecStart=/home/USERNAME/xray/xray run -config /home/USERNAME/xray/configs.json`
should be
`ExecStart=/home/USERNAME/xray/xray run -config /home/USERNAME/xray/config.json`
Fixed it. config.json without the s.

**Comment by houmie** [comment]
Awesome. Thank you for explaining this.

I'm actually using this script instead of downloading the zip file:

```bash -c "$(curl -L https://github.com/XTLS/Xray-install/raw/main/install-release.sh)" @ install --beta```

Because of the beta flag it downloads the latest `1.8.3` and creates the service automatically. So it makes it a lot easier I think.

In this case the binary is stored under `/usr/local/bin/xray` and I just checked `ls /usr/local/share/xray/` and I can see both geoip.dat  geosite.dat in there.  I think it's best that I put the iran.dat file also in that directory next to them.

What is your recommendations about how often they should be updated? Once a month?
Thanks

Nice one on the typo. As I wasn't using that service, I didn't notice it either. :-)

**Comment by SasukeFreestyle** [comment]
Yeah best Linux practices is to put binary files (xray) and assets in those folders, reason i chose not to was mostly to keep the guide simple without running into permission errors and "modifying" the system.
And if anything goes wrong you can simple just delete the xray folder.

You can use whatever is easier for you :)

My main goal of this guide is to spread awareness of the config file and its blocking rules.

I've seen 1000's of farsi guides on youtube and websites and here on github without the blocking rules on the server, and without any client routing rules.


Since the config file also has IP CIDR block, I usually only update every 2-3 months. 99,9% of all sites you will visit or block are in the iran.dat

**Comment by houmie** [comment]
Yes, it makes sense.  You're right, I haven't seen any tutorial about Iran with such details that you have provided. Thank you for that.

I have two more questions, please:

Looking at this example [here](https://cscot.pages.dev/2023/03/02/Xray-REALITY-tutorial/), the author has simply blocked all traffic to China within routing, including all ads.

```
"routing": {
        "domainStrategy": "IPIfNonMatch",
        "rules": [
            {
                "type": "field",
                "domain": [
                    "geosite:category-ads-all"
                ],
                "outboundTag": "block"
            },
            {
                "type": "field",
                "ip": [
                    "geoip:cn"
                ],
                "outboundTag": "block"
            }
        ]
    },
```

1) But I noticed you weren't satisfied by adding `"geoip:ir"` in there. You actually added all Iranian IP ranges. Is this because the geoip.ir is not extensive enough? I think you did that to be on the safe side.

2) I also noticed you mentioned some [additional Routings](https://github.com/SasukeFreestyle/XTLS-Iran-Reality#routing-rules) at the end of your tutorial. Please correct me if I'm wrong, but are Client rules  really that important?

> A) ext:iran.dat:ads in block section
> B) ext:iran.dat:ir in bypass section
> C) ext:iran.dat:other in bypass section


They come in three parts, 
A) blocking ads, 
B) bypassing Iranian ir domains, so that they connect directly instead of going over the proxy.
c) bypassing Iranian non-ir domains, same as above.

What is the worse case that no Client rule is defined?
- With the routing rule like this on the server side I would block all advertisements:
```
"routing": {
        "domainStrategy": "IPIfNonMatch",
        "rules": [
            {
                "type": "field",
                "domain": [
                    "geosite:category-ads-all"
                ],
                "outboundTag": "block"
            },
```
- And because all Iranian CIDR range plus geoip:ir are blocked, the user can't browse Iranian websites while the proxy is enabled. So the user has to disable the proxy, which is at worst just an inconvenience.  Other than that, no harm can be done if the client routing is not set.  Am I correct?

Many Thanks

**Comment by SasukeFreestyle** [comment]
> Yes, it makes sense. You're right, I haven't seen any tutorial about Iran with such details that you have provided. Thank you for that.

You're welcome :) Happy to help.

1. You guessed right, I found some ASN's that was missing in geoip.dat, then I got what you can call paranoid haha, and added all iranian CIDR just to be safe.

2. I consider client routing semi-important, its not necessary any "harm" in not using it, it would greatly benefit your internet speed when using Iranian websites with your Iranian IP, and from a internet security view, you are not letting ALL your traffic from the client (PC/Phone) be proxied to a single IP i.e the xray server and because you still are using your Iranian IP for Iranian websites your client now connects to multiple different IP's instead of just one IP (server).

With this you also don't have to disable the proxy, as you say its just an inconvenience to do it.

If everyone does routing it will become harder for firewall in Iran to ban or find your and others users xray servers. not impossible, but harder.

> With the routing rule like this on the server side I would block all advertisements:
You can block all advertisements in client or server. some users might like ads for some reason, so i decided that if users want it they can add that rule as a block in their client.

bootmortis has an excellent routing guide in farsi here https://github.com/bootmortis/iran-hosted-domains/blob/main/README.fa.md


In short. There are 3 rules a client and xray server can have, Direct, Proxy, Block.
In Server we block Iranian domains and CIDR with block rule, as a precaution if in-case of user not routing on client.
Client uses Direct rule (Iranian IP) for Iran based stuff.
Client uses proxy rule for everything else.

**Comment by houmie** [comment]
It makes sense. Thank you so much. 👍🏼
